File: Cpp/src/Point.cpp
//
// Created by aleksei on 05.04.2022.
//

#include "Point.h"
#include <QPainter>
#include <QMouseEvent>

Point::Point(QWidget *parent, unsigned size)
  : QWidget(parent)
  , m_size(size)
{
  setFixedWidth(static_cast<int>(m_size));
  setFixedHeight(static_cast<int>(m_size));
}

void Point::paintEvent(QPaintEvent *event)
{
  Q_UNUSED(event);
  QPainter painter(this);
  painter.setPen(QPen(Qt::black, m_size, Qt::SolidLine, Qt::RoundCap));

  painter.drawPoint(m_size / 2, m_size / 2);

}

void Point::moveEvent(QMoveEvent *event)
{
  emit positionChanged();
  QWidget::moveEvent(event);
}

File: Python/point.py
from PyQt5.QtWidgets import QWidget
from PyQt5 import QtGui
from PyQt5.QtCore import Qt, pyqtSignal
from PyQt5.QtGui import QPainter, QPen


class Point(QWidget):
    position_changed = pyqtSignal()
    def __init__(self, parent: QWidget = 0, size: int = 5):
        super(Point, self).__init__(parent)
        self._size = size
        self.setFixedWidth(self._size)
        self.setFixedHeight(self._size)

    def paintEvent(self, event: QtGui.QPaintEvent) -> None:
        painter = QPainter(self)
        painter.setPen(QPen(Qt.black, self._size, Qt.SolidLine, Qt.RoundCap))
        painter.drawPoint(self._size / 2, self._size / 2)

    def moveEvent(self, event: QtGui.QMoveEvent) -> None:
        self.position_changed.emit()
        super(Point, self).moveEvent(event)


